// mie-data-faker/src/documents/documents.js
import { faker } from '@faker-js/faker';
import moment from 'moment';


export function createDocuments(patIds, relationships, serviceLocations) {
    const docType = faker.helpers.arrayElement(relationships);
    return {
        pat_id: faker.helpers.arrayElement(patIds),
        doc_type: docType.doc_type,
        storage_type: docType.storage_type,
        service_location: faker.helpers.arrayElement(serviceLocations),
        origin_date: moment(faker.date.recent()).format('YYYY-MM-DD HH:mm'),
        ext_doc_id: faker.datatype.number({min: 10, max: 999999999}),
        interface: docType.interface,
        inpatient: faker.datatype.number({min: 0, max: 5}) > 1 ? 0 : 1,

    // commented out fields are those found in the appointments table but are not currently used

        // revision_number
        // user_id
        // origin_id
        // storage_id
        // service_date
        // enter_date - defaults to curent timestamp
        // revision_date - defaults to curent timestamp
        // approx_service_date
        // allow_for_teaching
    };
}
